lib/async/http/faraday/adapter.rb

```diff
@@ -148,6 +148,20 @@ def with_timeout(task: Async::Task.current)
 						yield
 					end
 				end
+				
+				def encoded_body(res)
+					body = res.read
+					return body if body.nil?
+					content_type = res.headers['content-type']
+					/\bcharset=\s*(.+?)\s*(;|$)/.match(content_type) do |match|
```

Comment:
This regular expression might be vulnerable to redos:

https://devina.io/redos-checker

Please check it.

Reply:
Fixed to avoid the abbrev.